feat(skills): Add N8N documentation search and workflow generation skills (#6)
inference-gateway/n8n-agent: CLAUDE.md

## Development Commands

```bash

> # Generate N8N node documentation (automatically installs ADL CLI if needed)
> task generate:docs
> 
> # Generate/regenerate code from ADL specification (runs generate:docs first)

Author: ```suggestion
# Generate/regenerate code from ADL specification
```

Reverting

## Adding New Functionality

### Skills Implementation
The following skills are currently defined:
- **search-n8n-docs**: Search through N8N node documentation to find relevant information about specific nodes, their parameters, and usage patterns
- **generate-n8n-workflow**: Generate complete N8N workflow YAML configurations based on user requirements, using documented nodes and best practices

To modify skills:
1. Update `agent.yaml` with skill definitions (ensure each skill has `id`, `name`, `type`, `tags`, and `schema` fields)
2. **REQUIRED**: Run `task generate` to regenerate the codebase after any skill changes (automatically installs ADL CLI)
3. Implement skill logic in generated skill files (look for TODO placeholders)
4. Write tests for each skill

**Critical Requirements for `task generate`:**
- The `task generate` command must be executed after adding or modifying skills in `agent.yaml`
- Skills require `id`, `name`, `type` (usually "function"), `tags` (array), and `schema` (JSON Schema) fields

> - ADL CLI is automatically installed via the `generate:docs` dependency

Author: ```suggestion
- ADL CLI is automatically installed via the `generate` task
```

- Docker
- Claude Code CLI

**ADL CLI Installation:**
- ADL CLI v0.18.2 is automatically installed when running `task generate:docs` or `task generate`
- Installation command: `curl -fsSL https://raw.githubusercontent.com/inference-gateway/adl-cli/main/install.sh | bash`
- Installs to `~/.local/bin/adl` and verifies PATH availability

Activate Flox with: `flox activate` (if Flox is installed)

## Important Constraints

- **Generated Files**: Never manually edit files with "DO NOT EDIT" headers
- **Configuration Changes**: Always modify `agent.yaml` and regenerate with `task generate`
- **ADL CLI Requirement**: ADL CLI is automatically installed when running `task generate`
- **Skills Validation**: All skills in `agent.yaml` must have `id`, `name`, `type`, `tags`, and `schema` fields
- **Generate Command Success**: The `task generate` command must complete successfully before development can proceed
- **Port Configuration**: Default 8080, configurable via `A2A_PORT` or `A2A_SERVER_PORT`

> ## Known Issues
> 
> ### ADL CLI Template Error
> The `task generate` command currently has a known template error:
> - **Error**: `failed to execute template ai/agents.md: template: template:70:15: executing "template" at <.Type>: can't evaluate field Type`
> - **Impact**: Most files generate successfully, but generation fails at the final template step
> - **Workaround**: The core functionality works - skill validation passes and main files are generated correctly
> - **Status**: This appears to be a bug in ADL CLI v0.18.2 template system rather than configuration issue
> 

Author: ```suggestion
```

Fixing it right now on the ADL CLI: https://github.com/inference-gateway/adl-cli/issues/59
